src/e3/anod/checkout.py

```diff
@@ -102,6 +107,50 @@ def update(
             )
         return result
 
+    def update_s3(
+        self, url: str, revision: str | None
+    ) -> tuple[ReturnValue, str, str]:  # all: no cover
+        """Update working dir using a S3 bucket as source.
+
+        :param url: s3 url to a s3 directory. s3://[profile@]bucket/suffix
+        :param revision: ignored
+        """
+        if os.path.isdir(self.working_dir):
+            old_commit = get_filetree_state(self.working_dir)
+        else:
+            old_commit = ""
+
+        # Extract profile from the url and perform basic
+        # url validation
+        m = re.match("s3://([a-zA-Z0-9_-]+@)?(.+)$", url)
```

Comment:
This is not a URL strictly speaking just a URI but I can try

Reply:
Not in the list of supported scheme

Reply:
See https://docs.python.org/3/library/urllib.parse.html

Reply:
```python
>>> from urllib.parse import urlparse
>>> url1 = "s3://profile@bucket/suffix"
>>> url2 = "s3://bucket/suffix"
>>> urlparse(url1)
ParseResult(scheme='s3', netloc='profile@bucket', path='/suffix', params='', query='', fragment='')
>>> urlparse(url2)
ParseResult(scheme='s3', netloc='bucket', path='/suffix', params='', query='', fragment='')
>>> 
```
Except for the `profile@` stuff it's not so bad. I suggested that just because the regular expression with authorized characters and so on, does not seem to be a good idea (to me).

Reply:
I have counter examples when you start using some weird character:

$ cat s3_keys_parsing
from urllib.parse import urlparse

print(urlparse("s3://bucket/some/prefix/objectwith&#13;carriagereturn"))

$ python ./s3_keys_parsing
ParseResult(scheme='s3', netloc='bucket', path='/some/prefix/objectwith&', params='', query='', fragment='13;carriagereturn')

You can use any UTF-8 character in an object key name. However, using certain characters in key names can cause problems with some applications and protocols. The following guidelines help you maximize compliance with DNS, web-safe characters, XML parsers, and other APIs.